src/lib.rs

```diff
@@ -401,6 +406,16 @@ impl Toast {
         self
     }
 
+    // HACK: f is static so that we know the function is valid to call.
+    //       this would be nice to remove at some point
+    pub fn on_activated(
+        mut self,
+        f: impl FnMut(&Option<ToastNotification>, &Option<IInspectable>) -> Result<()> + Send + 'static,
+    ) -> Self {
+        self.on_activated = Some(TypedEventHandler::new(f));
+        self
+    }
```

Comment:
There is no need to provide these arguments to the function, correct? I think we can just remove them and add them later if needed, also we could use `FnOnce` since that closure will only be activated once?

```suggestion
    // HACK: f is static so that we know the function is valid to call.
    //       this would be nice to remove at some point
    pub fn on_activated<F: FnOnce() -> Result<()> + Send + 'static>(
        mut self,
        f: F,
    ) -> Self {
        self.on_activated = Some(f);
        self
    }
```

you will also need to change `on_activated` signature

Reply:
As it turns out, the windows library adds the FnMut bound on all TypedEventHandlers.

Reply:
I kinda expected that, lets use `FnMut` then